packages/compiler/src/myst.ts

```diff
@@ -36,5 +59,106 @@ export function parsePanel(contents: string): Array<MystCard> {
   }
-  return splitCards;
+
+  let myPanel: MystPanel = { cards: splitCards };
+  if (panelStyle) {
+    myPanel = { ...myPanel, style: panelStyle };
+  }
+  return myPanel;
+}
+
+export function parseStyling(contents: string): [string, string] {
+  let yamlBlock = "";
+  let returnContents = "";
+  const contentLines = contents.split("\n");
+  if (contents.startsWith(":")) {
+    const yamlLines: Array<string> = [];
+    while (contentLines) {
+      if (!ltrim(contentLines[0]).startsWith(":")) {
+        break;
+      } else {
+        // TODO: Find way to fix non-null assertion warning
+        yamlLines.push(ltrim(contentLines.shift()!));
+      }
+    }
+    yamlBlock = yamlLines.join("\n");
+  }
+
+  returnContents = contentLines.join("\n").trim();
+  const returnPanel: [string, string] = [yamlBlock, returnContents];
+  return returnPanel;
+}
+
+export function parseYamlBlock(yamlBlock: string): Record<string, unknown> {
+  const styleContents = yamlBlock.split("\n");
+  const stylingRegex = /^:([a-z0-9]+):\s*([a-z\s\-0-9]+)/i;
+  // return object with key value pairs
+  let styles = {};
+  for (const line of styleContents) {
+    const stylingArray: RegExpExecArray | null = stylingRegex.exec(line);
+    if (stylingArray && stylingArray.length > 2) {
+      const key: string = stylingArray[1];
+      const value: string = stylingArray[2];
+      styles = { ...styles, [key]: value };
+    }
+  }
+  return styles;
+}
+
+function ltrim(rawString: string) {
+  return rawString.replace(/^\s+/gm, "");
+}
+
+export function mergeStyles(
+  panelStyle: MystStyling,
+  cardStyle: MystStyling
+): MystStyling {
+  let k: keyof MystStyling;
+  for (k in panelStyle) {
+    if (k in cardStyle) {
+      const panelPropDict = classesStringToKeyValues(panelStyle[k]!);
+      const cardPropDict = classesStringToKeyValues(cardStyle[k]!);
+      // merge Dicts
+      const mergedProps = { ...panelPropDict, ...cardPropDict };
+      cardStyle[k] = propDictToString(mergedProps);
+    } else {
+      cardStyle[k] = panelStyle[k];
+    }
+  }
+  return cardStyle;
+}
+
+export function classesStringToKeyValues(
+  classesString: string
+): Record<string, string> {
+  const htmlClasses = classesString.split(/\s+/);
+  let classDictionary = {};
+  for (const htmlClass of htmlClasses) {
+    const classArray = htmlClass.split("-");
+    let key, value;
+    // deal with single prop like "shadow"
+    if (classArray.length === 1) {
+      key = classArray[0];
+      value = "";
+    } else {
+      value = classArray.pop();
+      key = classArray.join("-");
+    }
+    classDictionary = { ...classDictionary, [key]: value };
+  }
+  return classDictionary;
+}
+
+function propDictToString(classObject: Record<string, string>): string {
+  let returnString = "";
+  for (const key in classObject) {
+    let concatValue;
+    if (classObject[key] === "") {
+      concatValue = key;
+    } else {
+      concatValue = [key, classObject[key]].join("-");
```

Comment:
Is this right? It looks likes that will be appending arrays to this structure, not sure if that's what we want?

Reply:
This returns a concatenated string (like ```col-lg-6``` or ```p-2```) that is then appended to the class string, I could use a ternary like above. The main issue if if there isn't a check for an empty string, it will append an unnecessary hyphen (i.e. ```shadow-``` rather than ```shadow```.